Move view handling logic to gen_inplace_or_view_type.py

ailzhang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #53343 Implement public API InferenceMode and its error handling
* #53342 Move as_view/increment_version to its separate key.
* **#53341 Move view handling logic to gen_inplace_or_view_type.py**
* #53340 Move common autograd utils functions from gen_variable_type.py to api/autograd.py.

Differential Revision: [D26973912](https://our.internmc.facebook.com/intern/diff/D26973912)

Commit:
Move view handling logic to gen_inplace_or_view_type.py

[ghstack-poisoned]

Commit:
Update on "Move view handling logic to gen_inplace_or_view_type.py"




[ghstack-poisoned]

Commit:
Update on "Move view handling logic to gen_inplace_or_view_type.py"




[ghstack-poisoned]

Commit:
Update on "Move view handling logic to gen_inplace_or_view_type.py"




[ghstack-poisoned]

Commit:
Update on "Move view handling logic to gen_inplace_or_view_type.py"




[ghstack-poisoned]

ailzhang commented:
Test Plan: verified the generated VariableTypeEverything.cpp are the same after this PR.

albanD commented on tools/autograd/gen_autograd.py:
```diff
@@ -4,6 +4,7 @@
 
 python -m tools.autograd.gen_autograd \
        build/aten/src/ATen/Declarations.yaml \
+       aten/src/ATen/native/native_functions.yaml \
```
Why is that? Is this something you added in this PR? Or that was added before and the doc wasn't updated?

ailzhang replied:
It was added before the PR but doc wasn't updated :P

albanD replied:
ok ok
Thanks for fixing that then!

albanD reviewed (approved):
Looks good!

It is so much easier to review simple refactor when there is no feature mixed into it :) Thanks for taking the time to do this!

Commit:
Update on "Move view handling logic to gen_inplace_or_view_type.py"




[ghstack-poisoned]

ailzhang commented on tools/autograd/gen_inplace_or_view_type.py:
```diff
@@ -0,0 +1,268 @@
+from tools.codegen.api import cpp
+from tools.codegen.api.autograd import *
```
I can fix all the new ones introduced in this PR and send out a followup PR to clean up the folder later ;) 
But I didn't find a way to add this in flake8 so it'll be hard to prevent future changes :P

Commit:
Update on "Move view handling logic to gen_inplace_or_view_type.py"




[ghstack-poisoned]

Commit:
Update on "Move view handling logic to gen_inplace_or_view_type.py"


Differential Revision: [D26973912](https://our.internmc.facebook.com/intern/diff/D26973912)

[ghstack-poisoned]
